lib/fml_forms/validations.rb

```diff
@@ -40,35 +47,99 @@ def initialize(field, data, form)
       # strip the ! if there was one
       data = data[1..-1] if @negative
 
-      @child = field
+      @field = field
       @parent = form.fields[data]
     end
 
+    def conforms!
+      if ["yes_no", "checkbox"].index(@parent.type).nil?
+        raise InvalidSpec.new(<<-EOM)
+Field #{@field.name} depends on field #{@parent.name}, which is not a boolean.
+Fields may only depend on "yes_no" or "checkbox" fields, but #{@parent.name} is a
+"#{@parent.type}" field.
+      EOM
+      end
+    end
+
+    def required?
+      @parent.visible? && (@parent.value) == @required
+    end
+
+    def valid?
+      !(required? && @field.empty?)
+    end
+
     def validate
       # if parent is @required, child must be non-empty. Note that @parent is
       # required to be a boolean element, so we don't need to worry about ""
       # being a truthy value
-      if @parent.value == @required && @child.empty?
+      if not valid?
         debug_message = <<-EOM
-Field #{@child.name}:#{@child.value.inspect} must be present when #{@parent.name}:#{@parent.value.inspect} is #{@required}
+Field #{@field.name}:#{@field.value.inspect} must be present when #{@parent.name}:#{@parent.value.inspect} is #{@required}
         EOM
         user_message = "This field is required"
-        err = DependencyError.new(user_message, debug_message, @child.name, @parent.name)
-        @child.errors << err
+        err = DependencyError.new(user_message, debug_message, @field.name, @parent.name)
+        @field.errors << err
         raise err
       end
     end
   end
 
-  class MinLengthValidation
+  class RequiredIfTextEquals < BaseValidation
     def initialize(field, data, form)
       @field = field
-      @minlength = data
+      @wanted_values = Array(data['value'] || data['values'])
+      @parent = form.fields[data['field']]
+    end
+
+    def conforms!
+      if ["select", "text"].index(@parent.type).nil?
+        raise InvalidSpec.new(<<-EOM)
+Field #{@field.name} depends on field #{@parent.name}, which is not a boolean.
+Fields may only depend on "select" or "text" fields, but #{@parent.name} is a
+"#{@parent.type}" field.
+      EOM
+      end
+    end
+
+    def required?
+      @wanted_values.include?(@parent.value)
+    end
+
+    def valid?
+      !(required? && @field.empty?)
     end
 
     def validate
+      if not valid?
+        debug_message = <<-EOM
+Field #{@field.name}:#{@field.value.inspect} must be #{@wanted_values}
+        EOM
+        user_message = "when #{@parent.name} is '#{@wanted_values}', #{@field.name} must be filled in"
+        err = ValidationError.new(user_message, debug_message, @field.name)
+        @field.errors << err
+        raise err
+      end
+    end
+  end
+
+  class MinLengthValidation < BaseValidation
+    def initialize(field, data, form)
+      @field = field
+      @minlength = data
+    end
+
+    def required?
+      false
```

Comment:
I think this needs to be defined because it's called in other places, but it probably shouldn't just return `true`. Likely an oversight from early spiking and testing.